internal/restic/snapshot.go

```diff
@@ -27,17 +27,45 @@ type Snapshot struct {
 	id *ID // plaintext ID, used during restore
 }
 
+func pathSplit(path string) (root string, lst []string) {
+	for root = filepath.Clean(path); ; {
+		var file string
+		root, file = filepath.Split(root)
+		if file == "" {
+			break
+		}
+		lst = append([]string{file}, lst...)
+	}
+	return
+}
+
+func pathMangle(path string, prefix string, strip int) string {
```

Comment:
We need to find a better name for this function, and it also needs extensive tests. What about `pathStripPrefix`?

Reply:
pathStripPrefix is okay, I'll rename it.